bundles/org.openhab.binding.flicbutton/src/main/java/org/openhab/binding/flicbutton/internal/discovery/FlicSimpleclientDiscoveryServiceImpl.java

```diff
@@ -0,0 +1,136 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.flicbutton.internal.discovery;
+
+import java.io.IOException;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.openhab.binding.flicbutton.FlicButtonBindingConstants;
+import org.openhab.binding.flicbutton.internal.FlicButtonHandlerFactory;
+import org.openhab.binding.flicbutton.internal.util.FlicButtonUtils;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.flic.fliclib.javaclient.Bdaddr;
+import io.flic.fliclib.javaclient.FlicClient;
+import io.flic.fliclib.javaclient.GeneralCallbacks;
+import io.flic.fliclib.javaclient.GetInfoResponseCallback;
+import io.flic.fliclib.javaclient.enums.BdAddrType;
+import io.flic.fliclib.javaclient.enums.BluetoothControllerState;
+
+/**
+ * For each configured flicd service, there is a {@link FlicSimpleclientDiscoveryServiceImpl} which will be initialized
+ * by
+ * {@link FlicButtonHandlerFactory}.
+ *
+ * It can scan for Flic Buttons already that are already added to fliclib-linux-hci ("verified" buttons), *
+ * but it does not support adding and verify new buttons on it's own.
+ * New buttons have to be added (verified) e.g. via simpleclient by Shortcut Labs.
+ * Background discovery listens for new buttons that are getting verified.
+ *
+ * @author Patrick Fink - Initial contribution
+ */
+public class FlicSimpleclientDiscoveryServiceImpl extends AbstractDiscoveryService
+        implements FlicButtonDiscoveryService {
+    private final Logger logger = LoggerFactory.getLogger(FlicSimpleclientDiscoveryServiceImpl.class);
+
+    private boolean activated = false;
+    private ThingUID bridgeUID;
+    private FlicClient flicClient;
+
+    public FlicSimpleclientDiscoveryServiceImpl(@NonNull ThingUID bridgeUID) {
+        super(FlicButtonBindingConstants.SUPPORTED_THING_TYPES_UIDS, 2, true);
+        this.bridgeUID = bridgeUID;
+    }
+
+    @Override
+    public void activate(@NonNull FlicClient flicClient) {
+        this.flicClient = flicClient;
+        activated = true;
+        super.activate(null);
+    }
+
+    @Override
+    public void deactivate() {
+        activated = false;
+        super.deactivate();
+    }
+
+    @Override
+    protected void startScan() {
+        try {
+            if (activated) {
+                discoverVerifiedButtons();
+            }
+
+        } catch (IOException e) {
+            logger.warn("Error occured during button discovery", e);
```

Comment:
Are you logging the stack trace by intention?

Reply:
Yes. Under normal conditions, this is extremely unlikely to happen, so it's good to have the exception then.